src/settings/profile.hpp

```diff
@@ -80,6 +83,26 @@ class Profile
         return true;
     }
 
+    void move_per_file_settings(const int original_index, const int new_index)
+    {
+        if (original_index == new_index)
+            return;
+
+        if (original_index < new_index)
+        {
+            auto old_it = per_file_settings_.begin() + original_index;
+            auto new_it = per_file_settings_.begin() + new_index;
+            std::rotate(old_it, old_it + 1, new_it);
+        }
+        else
+        {
+            const auto len = per_file_settings_.size();
+            auto old_it    = per_file_settings_.rbegin() + (len - original_index - 1);
+            auto new_it    = per_file_settings_.rbegin() + (len - new_index);
+            std::rotate(old_it, old_it + 1, new_it);
+        }
```

Comment:
This logic is better for performance, but I don't think we need it for such a short list
Please either use `erase` and `insert` or extract this logic into a utility function